### ISSUE TITLE
Python Uninstalls when closing VS Code

### ISSUE THREAD

**Issue by bjone161** [microsoft/vscode#190044]
Type: <b>Bug</b>

When I close and then re-open VS Code, I need to re-install Python every time. It is the only extension that has the issue. 

VS Code version: Code 1.81.0 (6445d93c81ebe42c4cbd7a60712e0b17d9463e97, 2023-08-02T12:37:13.485Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6360U CPU @ 2.00GHz (4 x 1992)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (18.88GB free)|
|Process Argv|--user-data-dir C:\\Users\\jone8\\AppData\\Roaming\\.anaconda\\navigator\\Code --crash-reporter-id 64bf1697-86c7-496b-8eca-90e2c05e143c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
chatgpt-gpt4-gpt3-vscode|Eas|1.0.9
vscode-docker|ms-|1.26.0
vscode-pylance|ms-|2023.8.10
jupyter|ms-|2023.7.1002162226
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30803844
3biah626:30602489
f6dab269:30613381
showlangstatbar:30737416
vsctsb:30748421
a2ce3375:30757347
7ij38806:30736111
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofa:30804715
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
e537b577:30795824
dsvsc013:30795093
dsvsc014:30804076

```

</details>

**Comment by xBeho1der** [comment]
> Type: Bug
> 
> When I close and then re-open VS Code, I need to re-install Python every time. It is the only extension that has the issue.
> 
> VS Code version: Code 1.81.0 ([6445d93](https://github.com/microsoft/vscode/commit/6445d93c81ebe42c4cbd7a60712e0b17d9463e97), 2023-08-02T12:37:13.485Z) OS version: Windows_NT x64 10.0.19045 Modes:
> 
> System Info
> Extensions (8)
> A/B Experiments

I have met the same situation. #190051

**Comment by ladyrick** [comment]
+1

**Comment by bjone161** [comment]
No other profiles were used

**Comment by aaronjnewman** [comment]
I am also having similar issues. They seem to have started with the update to v 1.81.0.

First off, when I open a Python file in VS code (specifically a Jupyter notebook running a Python kernel) that previously worked fine, VS Code tells me I need to install the Python and Jupyter extensions (which were previously installed). Clicking the "install" buttons on those popups has no effect however. Trying to install through the extensions manager leads to an "installing" message but never completes. On force-quit, I need to go through the same cycle again, and the Python extension never installs.

Thanks to this thread, I tried switching from the Default profile to either of the two custom profiles I had previously set up. Once I switched to a pre-existing profile, I still had to re-install the Python and Jupyter extensions, but at least the install completed. And now, when I quite & restart VS Code, the extensions remain installed.

However, if I switch back to the Default profile, the problems re-occur. My guess is that the latest release has a bug related to the default profile that is interfering with (at least) the Python and Jupyter extensions. I also suspect that users who had not set up non-Default profiles will be stuck until this bug is fixed.

**Comment by sean-mcmanus** [comment]
I figured out a repro -- install a vsix with the same version as another vsix that has been updated. Once the user is in that state, it appears to the user that the extension is uninstalled (i.e. no version issue), but when they install it again and reload, VS Code will remove the extension again because it's still thinking that the newly installed version is the one it's supposed to uninstall. The workaround to fix is this is to install a vsix with a different version. Another report appears to be at https://github.com/microsoft/vscode/issues/190051 .
